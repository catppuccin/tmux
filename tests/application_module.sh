#!/usr/bin/env bash

script_dir=$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd -P)
# shellcheck disable=SC1091
source "${script_dir}/helpers.sh"

# Tests that the default options are set correctly
tmux source "${script_dir}/../catppuccin_options_tmux.conf"
tmux source "${script_dir}/../catppuccin_tmux.conf"

application_text=$(get_option @catppuccin_application_text)

print_option @catppuccin_status_application | grep -q "$application_text" ||
  echo "@catppuccin_status_application expanded @catppuccin_application_text more than once"

print_option @catppuccin_status_application | grep -q "@thm_" &&
  echo "@catppuccin_status_application did not expand all colors"

print_option E:@catppuccin_status_application | sed -E 's/(bash|fish|zsh)/<application>/'
