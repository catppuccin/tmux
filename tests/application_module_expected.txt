@catppuccin_status_application #[fg=#eba0ac]#[fg=#11111b,bg=#eba0ac] #[fg=#cdd6f4,bg=#313244]#{E:@catppuccin_application_text}#[reverse]█#[noreverse]